binutils-build/Makefile

```diff
@@ -2,60 +2,93 @@
 # Makefile to compile binutils cross
 #
 # Use SRC_DIR to alter the path of the source (default ../binutils)
-# Use VERSION to specify the version (alters the name of the created drawer)
 #
+# Use BINUTILS_VERSION to specify the version (alters the name of the
+# created drawer)
+
+ifeq ($(BINUTILS_VERSION),)
+$(error BINUTILS_VERSION is a required variable)
+endif
 
 SRC_DIR=../binutils/repo
-VERSION=2.23.2
-CROSS_BUILD_DIR=cross-$(VERSION)
-NATIVE_BUILD_DIR=cross-$(VERSION)
+CROSS_BUILD_DIR=cross-$(BINUTILS_VERSION)
+NATIVE_BUILD_DIR=cross-$(BINUTILS_VERSION)
 PREFIX=/usr/local/amiga
 REAL_SRC_DIR=$(realpath $(SRC_DIR))
 
-all: cross
+############################
+### BEGIN: SETUP OPTIONS ###
+############################
+ifeq ($(BINUTILS_VERSION),2.23.2)
+CROSS_CFLAGS=CFLAGS="					\
+				-Wno-switch		\
+				-Wno-unused		\
+							"
+CROSS_CONFIGURE_ARGUMENTS=				\
+				--enable-plugins	\
+				--target=ppc-amigaos	\
+				--prefix=$(PREFIX)
+NATIVE_CFLAGS=CFLAGS="					\
+				-Wno-switch		\
+				-Wno-unused		\
+							"
+NATIVE_LDFLAGS=LDFLAGS="				\
+				-lunix			\
+							"
+NATIVE_CONFIGURE_ARGUMENTS=				\
+				--target=ppc-amigaos	\
+				--host=ppc-amigaos	\
+				--disable-nls		\
+				--prefix=/gcc
+else
+ifneq ($(BINUTILS_VERSION),2.40)
+$(error Unexpected binutils version)
+else
+CROSS_CFLAGS=CFLAGS="					\
+				-Wno-sign-compare	\
+							"
+CROSS_CONFIGURE_ARGUMENTS=				\
+				--disable-plugins	\
+				--disable-gdb		\
+				--disable-sim		\
+				--target=ppc-amigaos	\
+				--prefix=$(PREFIX)
+NATIVE_CFLAGS=CFLAGS="					\
+				-mcrt=clib4		\
+				-Wno-sign-compare	\
+							"
+NATIVE_LDFLAGS=LDFLAGS="				\
+				-mcrt=clib4		\
+							"
+NATIVE_CONFIGURE_ARGUMENTS=				\
+				--target=ppc-amigaos	\
+				--host=ppc-amigaos	\
+				--disable-plugins	\
+				--disable-gdb		\
+				--disable-sim		\
+				--disable-nls		\
+				--prefix=/gcc
+endif
+endif
+############################
+### ENDER: SETUP OPTIONS ###
+############################
 
-cross-2.23.2:
-	mkdir -p $(CROSS_BUILD_DIR)
-	cd $(CROSS_BUILD_DIR); CFLAGS="-Wno-switch -Wno-unused" $(REAL_SRC_DIR)/configure \
-									--enable-plugins \
-									--target=ppc-amigaos \
-									--prefix=$(PREFIX)
-	make -C $(CROSS_BUILD_DIR)
+.PHONY: cross native
 
-cross-2.40:
+cross:
 	mkdir -p $(CROSS_BUILD_DIR)
-	cd $(CROSS_BUILD_DIR); CFLAGS="-Wno-sign-compare" $(REAL_SRC_DIR)/configure \
-									--disable-plugins \
-									--disable-gdb \
-									--disable-sim \
-									--target=ppc-amigaos \
-									--prefix=$(PREFIX)
-	make -C $(CROSS_BUILD_DIR)
+	cd $(CROSS_BUILD_DIR);			\
```

Comment:
Possibly that is not needed since the execution of the make a few lines below uses the `-C` argument that point in which folder the makefile is and where it should be executed.

Reply:
In the recipe for `cross`, for example:
```
cross:
	mkdir -p $(CROSS_BUILD_DIR)
	cd $(CROSS_BUILD_DIR);			\
		$(CROSS_CFLAGS)			\
		$(REAL_SRC_DIR)/configure	\
		$(CROSS_CONFIGURE_ARGUMENTS)
	$(MAKE) -C $(CROSS_BUILD_DIR)
```
there are three step.
1 - Create `$(CROSS_BUILD_DIR)`
2 - CD into it and then invoke configure from the "real" directory so that it generates files in `$(CROSS_BUILD_DIR)`
3 - Finally, invoke the standard makefile in that `$(CROSS_BUILD_DIR)`

If we removed step 2, autotools would end up generating the makefiles in `../$(CROSS_BUILD_DIR)`